### DISCUSSION TITLE
Data Fields Distribution - Visual aspects

### DISCUSSION THREAD

**Discussion by ea5wa** [ea4k/klog#203] [Show and tell]
Sometimes I don't really like how data is distributed in the KLog windows. I'm trying to learn about layouts and stretchers in Qt in order to improve the visual aspect. I'm starting to make some changes in QSO tab, I had pushed up name and RX power a little bit.

![imagen](https://user-images.githubusercontent.com/54686689/120883858-e7aa2b00-c5df-11eb-90b5-96000a665b5f.png)

I want to modify some textboxes widths....

What do you thing about that? Really matters?

73 de EA5WA Juan Carlos

**Comment by ea4k** [comment]
The UI is VERY important. 
A bad UI can make users to leave KLog whatever is in the background.

Suggestions for the UI are more than welcome!

**Reply by foldynl** [reply]
I agree. An UI should be intuitive and simple - allow access from the keyboard as much as possible without jumping from keyboard to mouse and back.
I have following comments to the main window:
- Issue #265 
- Issue #266 
- Issue #268 
- to be honest, it is unnecessary to have two fields for frequency when split mode is not active; In case of unchecked split mode, the second frequency field should not be present or disabled.
- Name field is too long - Locator and Name can be side by side.
- RST field is too long.
- Comment Tab - the fields should be present in the QSO tab - not in separate tab.

I hope that my comments will be useful for you.

I am glad that you are actively developing KLog. Currently, I'm using another program on Linux, but it's not so actively developed. Unfortunately, some promising logs are dead for several years.

Thank you
73 de OK1MLG Ladislav

**Comment by ea5wa** [comment]
Hi Ladislav:

All your comments are very interesting. 

I'm learning how to create User Interface, but I'm just a newbie..... I'll try to make some tests in a few days, but now I'm working on the Dark Mode for KLog.

73 de EA5WA Juan Carlos

**Comment by foldynl** [comment]
Hi Juan Carlos,

I am a programmer but only specialist for back-ends. So I have just a limited knowledge about the UI design. I will try to help as best I can.

You do good job.

**Comment by ea5wa** [comment]
Some tests using GridLayout...

![image](https://user-images.githubusercontent.com/54686689/123293778-8f968280-d514-11eb-8a86-68801608b948.png)

**Reply by foldynl** [reply]
I thought about your design of QSO window. I am attaching an alternative layout for it. I can provide also UI file (QT Designer) for it. I am attaching two pictures to show how the layout look like when resized.  

![qso_new3](https://user-images.githubusercontent.com/942769/123472507-8df1bb00-d5f7-11eb-94cc-644ef7847dca.png)
![qso_new3_large](https://user-images.githubusercontent.com/942769/123472510-8e8a5180-d5f7-11eb-8e63-3d27325b57a9.png)

**Comment by ea5wa** [comment]
I would like to test it in Linux. Could you upload it to github?
My test was ok in windows but last morning I tested in Linux and don't like it....

**Reply by foldynl** [reply]
It was just an concept - a quick work in QT Designer without any special setting of layouts or parameters etc. I have exported the source code to git so you can pull it now.
https://github.com/foldynl/QSO.git

**Comment by ea5wa** [comment]
Thanks Ladislav. I'll take a look tomorrow
